### ISSUE TITLE
Code should be cleaned.

### ISSUE THREAD

**Issue by xpheres** [xpheres/analyticaltranslator#14]
At this point, the program is big enough to start thinking seriously about cleaning it. 
It is a priority to make the code readable, scalable and modular and to apply good style programming rules and maybe choose new methods to resolve the problems involved in this software.

I asked for help and tips regarding this and I obtained the following tips:

anonymized: xpheres, mysql_connect()   <-- stop using that. HEre is why: It is Deprecated. Doesn't support new mysql stuff. Will be removed from PHP.

anonymized: xpheres, use PDO to connect to mysql instead: http://wiki.hashphp.org/PDO_Tutorial_for_MySQL_Developers

anonymized: xpheres, http://thisinterestsme.com/php-best-practises/  <-- read #5 on what to do about cleaning up your code. :)

anonymized: xpheres, template advice: http://www.phptherightway.com/#templating

anonymized: xpheres, now where you are on that php-best-practices blog page, might as well look through the other good practices. ;)

anonymized: xpheres: have a look at pdo and twig first. you know you can exclude css into own file but you doesn't cause it makes sense to write that less css inline? thats good. but use css for all formatting, no <font> tags

anonymized: xpheres, also, when you are done, run the HTML (the result of the webpage) through this validator: http://validator.w3.org/

anonymized: xpheres, as for translator, you might want to make use of this: http://php.net/manual/en/messageformatter.formatmessage.php   <----  http://php.net/manual/en/book.intl.php MessageFormatter, NumberFormatter. and etc.

**Comment by Putti** [comment]
Translator*.html files could be fitted to one index.html file. There is so much common elements between those files so it seems logical to me. If everything can't be done by just using html we could use for example php (http://www.phptherightway.com/#templating) or other programmatic way to manipulate which items are shown. What do you say?

**Comment by xpheres** [comment]
Absolutely! I started this process with the HTML code, there is so much code repeated that can be reused.

Please check my last commit: https://github.com/xpheres/analyticaltranslator/commit/50f1961159d6618c26794dae78379b9ed728ff2e

I used this method to append and reuse HTML code: http://www.hongkiat.com/blog/html-import/

I have to do something similar to the php code. The code should be modular and reusable if I want it to progress and to allow people to understand it.

**Comment by Putti** [comment]
I took a look to those files you commented. If we use different html files for different elements of the site then I think it would be (maybe!) better to use some library like [Polymer project](https://www.polymer-project.org/0.5/) to handle the imports. The technique in the recent commit made it difficult to follow the content which comes from querySelector() call, and also it didn't(?) solve the code repetition part because now those querySelector() function calls need to be done. 

As one suggestion, in polymer the whole bootstrap menu could be imported with one tag which could be, for example, \<bootstrap-element\>. This has still the same problem as the latest commits way: you need to check what is in that other html file and so it becomes also difficult to see what content is really there behind the \<bootstrap-element\> tag. But in Polymer you don't need to put the whole menu in one tag but you can instead chop it in smaller elements and then import those elements using html tags, so that would be an intuitive way and you'd see what kind of content is in there.

I try to look still if there is some other Polymer like libraries which could be even better for this job. And do you know if that technique you used can import the whole html file instead of one tag? I took a quick peek on it now, and guess it can so why not to import the whole html file at one time? 

We should maybe now first decide the way(s) of trying to minimize the code reuse and readability, and then, after we have a conclusion, implement it?

**Comment by Putti** [comment]
Can the code after "//We import the bootstrap menus" be put to one querySelector() call with this technique? querySelector() calls look OK to me but trying to minimize the amount of those would be something I think needs to be done.

Other way of using html imports could be by using an external library. [Polymer project](https://www.polymer-project.org) is one such library which enables the use of html imports by creating custom html tags. For readability I think it would be better because there is no need for JavaScript when importing the content from other html files (only \<custom-element\> kind of tags).

**Comment by xpheres** [comment]
Please check last commit: https://github.com/xpheres/analyticaltranslator/commit/fe75eae30ede4d70c1cb05cd726fc43e783a64b1

Now the whole html is imported at once.

**Comment by xpheres** [comment]
Polymer looks great, I never used it so it will take me time to get used to it.

**Comment by Putti** [comment]
Commit fe75eae looked awesome! Now when some changes are done to that menu, no reference changes are needed. :)

I recommend definitely to try out Polymer as if it might suit to this project.

**Comment by xpheres** [comment]
Everything seems to work fine excepting the rotating quotes that does not change anymore... I have to find out why...

**Comment by Putti** [comment]
What do you mean by rotating quotes? I cleaned some css so I might have deleted that feature accidentally. Or did it happen after the 50f1961 commit?

**Comment by xpheres** [comment]
I mean the text at the bottom that should be changing, is repeating now all the time, but it is not important, I will find out and correct it :)
